[BUG] NexusIQScan Plugin 2.0.8 doesn't work

**Describe the bug**
The nexusIQScan plugin execute failed in Jenkins pipeline.
We are trying to setup the nexusIQSacn in a two build.gradle setup, a master and a child. However it is not working. 

Below are the logs from Jenkins pipeline:
2021-04-27T05:51:53.939+0000 [INFO] [com.android.build.gradle.internal.profile.RecordingBuildListener] Task :nexusIQScan in app_bt Finished
2021-04-27T05:51:53.953+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Completing Build operation 'Task :nexusIQScan'
2021-04-27T05:51:53.953+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Build operation 'Task :nexusIQScan' completed
2021-04-27T05:51:53.954+0000 [INFO] [org.gradle.execution.plan.DefaultPlanExecutor] :nexusIQScan (Thread[Execution worker for ':' Thread 2,5,main]) completed. Took 1 mins 3.694 secs.
2021-04-27T05:51:53.954+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :nexusIQScan failed
2021-04-27T05:51:53.954+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :app:nexusIQScan completed, executed: false
2021-04-27T05:51:53.955+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :nexusIQScan completed, executed: true
2021-04-27T05:51:53.955+0000 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] 
.
.
2021-04-27T05:51:54.069+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2021-04-27T05:51:54.069+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] FAILURE: Build failed with an exception.
2021-04-27T05:51:54.070+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2021-04-27T05:51:54.070+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * What went wrong:
2021-04-27T05:51:54.070+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Execution failed for task ':nexusIQScan'.
2021-04-27T05:51:54.070+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > Could not scan the project: null
2021-04-27T05:51:54.070+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2021-04-27T05:51:54.070+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Try:
2021-04-27T05:51:54.070+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]  Run with --scan to get full insights.
2021-04-27T05:51:54.071+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2021-04-27T05:51:54.071+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Exception is:
2021-04-27T05:51:54.072+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':nexusIQScan'.
2021-04-27T05:51:54.072+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:205)
2021-04-27T05:51:54.072+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
2021-04-27T05:51:54.072+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:203)

**To Reproduce**
Steps to reproduce the behavior:
1. Create a repo with a two build.gradle setup, a master and a child. 

2. In the master build.gradle declare the nexus plugin:
//Sonatype IQ Plugin
plugins { 
    id "org.sonatype.gradle.plugins.scan" version "2.0.8" // Update the version as needed 
}

3. In the child build.gradle, apply nexus plugin and configure IQ Server settings 
 //apply sonatype plugin
apply plugin: 'org.sonatype.gradle.plugins.scan'
//NexusIQ Configuration 
nexusIQScan {
    username = 'admin' // Make sure to use an user with the role 'Application Evaluator' in the given IQ Server application
    password = 'pass'
    serverUrl = 'http://localhost:8070'
    applicationId = 'app'
    stage = 'build' // build is used if omitted
    allConfigurations = true // if true includes the dependencies in all resolvable configurations. By default is false, meaning only 'compileClasspath' and 'releaseCompileClasspath' are considered
    resultFilePath = 'results.json' // Optional. JSON file containing results of the evaluation
}

4. In Jenkins pipeline, run the following command:
./gradlew nexusIQScan

5. See pipeline logs


**Expected behavior**
Able to scan the OSS in the Application code
View the report in Nexus IQ WebUI


**Desktop (please complete the following information):**
 - OS: Linux
 - Gradle Version: 6.1.1
 - JVM Version and Flavor: Java8
 - Nexus IQ Server Version: 97

> Also @puash:
> 
> > In the master build.gradle declare the nexus plugin
> 
> > In the child build.gradle, apply nexus plugin and configure IQ Server settings
> 
> The plugin can be applied and configured directly on the root project (or master as you call it), is there a reason why you're applying this on the "child" project?

Hi @guillermo-varela ,
Sorry I made a mistake in the steps to reproduce in my first post. The error msg produced when I apply and configure the plugin in the root project build.gradle file.

I tested 2 methods using the plugin (version 1.2.3 -2.0.8) however non of it works.

**Method (1): Apply and configure the plugin in root project (master)**
> In the master build.gradle declare the nexus plugin:
> //Sonatype IQ Plugin
> plugins {
> id "org.sonatype.gradle.plugins.scan" version "2.0.8" // Update the version as needed
> }
> //apply sonatype plugin
> apply plugin: 'org.sonatype.gradle.plugins.scan'
> //NexusIQ Configuration
> nexusIQScan {
> username = 'admin' // Make sure to use an user with the role 'Application Evaluator' in the given IQ Server application
> password = 'pass'
> serverUrl = 'http://localhost:8070'
> applicationId = 'app'
> stage = 'build' // build is used if omitted
> allConfigurations = true // if true includes the dependencies in all resolvable configurations. By default is false, meaning only 'compileClasspath' and 'releaseCompileClasspath' are considered
> resultFilePath = 'results.json' // Optional. JSON file containing results of the evaluation
> }

**Method (2): Apply the plugin in root project (master) and configure it in the app project (child)**
> In the master build.gradle declare the nexus plugin:
> //Sonatype IQ Plugin
> plugins {
> id "org.sonatype.gradle.plugins.scan" version "2.0.8" // Update the version as needed
> }
> 
> In the child build.gradle, apply nexus plugin and configure IQ Server settings
> //apply sonatype plugin
> apply plugin: 'org.sonatype.gradle.plugins.scan'
> //NexusIQ Configuration
> nexusIQScan {
> username = 'admin' // Make sure to use an user with the role 'Application Evaluator' in the given IQ Server application
> password = 'pass'
> serverUrl = 'http://localhost:8070'
> applicationId = 'app'
> stage = 'build' // build is used if omitted
> allConfigurations = true // if true includes the dependencies in all resolvable configurations. By default is false, meaning only 'compileClasspath' and 'releaseCompileClasspath' are considered
> resultFilePath = 'results.json' // Optional. JSON file containing results of the evaluation
> }

Method (1) error logs:
2021-04-27T05:51:53.939+0000 [INFO] [com.android.build.gradle.internal.profile.RecordingBuildListener] Task :nexusIQScan in app_bt Finished
2021-04-27T05:51:53.953+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Completing Build operation 'Task :nexusIQScan'
2021-04-27T05:51:53.953+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Build operation 'Task :nexusIQScan' completed
2021-04-27T05:51:53.954+0000 [INFO] [org.gradle.execution.plan.DefaultPlanExecutor] :nexusIQScan (Thread[Execution worker for ':' Thread 2,5,main]) completed. Took 1 mins 3.694 secs.
2021-04-27T05:51:53.954+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :nexusIQScan failed
2021-04-27T05:51:53.954+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :app:nexusIQScan completed, executed: false
2021-04-27T05:51:53.955+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :nexusIQScan completed, executed: true
2021-04-27T05:51:53.955+0000 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock]
.
.
2021-04-27T05:51:54.069+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
2021-04-27T05:51:54.069+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] FAILURE: Build failed with an exception.
2021-04-27T05:51:54.070+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
2021-04-27T05:51:54.070+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * What went wrong:
2021-04-27T05:51:54.070+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Execution failed for task ':nexusIQScan'.
2021-04-27T05:51:54.070+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > Could not scan the project: null
2021-04-27T05:51:54.070+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
2021-04-27T05:51:54.070+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Try:
2021-04-27T05:51:54.070+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Run with --scan to get full insights.
2021-04-27T05:51:54.071+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
2021-04-27T05:51:54.071+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Exception is:
2021-04-27T05:51:54.072+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':nexusIQScan'.
2021-04-27T05:51:54.072+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:205)
2021-04-27T05:51:54.072+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
2021-04-27T05:51:54.072+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:203)

Method (2) error logs:
2021-04-27T07:48:46.188+0000 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] 
2021-04-27T07:48:46.188+0000 [LIFECYCLE] [class org.gradle.internal.buildevents.TaskExecutionLogger] > Task :nexusIQScan FAILED
2021-04-27T07:48:46.188+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Build operation 'Task :nexusIQScan' started
2021-04-27T07:48:46.199+0000 [INFO] [com.android.build.gradle.internal.profile.RecordingBuildListener] Task :nexusIQScan in app_bt Starting
2021-04-27T07:48:46.224+0000 [DEBUG] [org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter] Putting task artifact state for task ':nexusIQScan' into context took 0.021 secs.
2021-04-27T07:48:46.239+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Build operation 'Snapshot task inputs for :nexusIQScan' started
2021-04-27T07:48:46.252+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Completing Build operation 'Snapshot task inputs for :nexusIQScan'
2021-04-27T07:48:46.252+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Build operation 'Snapshot task inputs for :nexusIQScan' completed
2021-04-27T07:48:46.252+0000 [DEBUG] [org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter] Removed task artifact state for task ':nexusIQScan' from context.
2021-04-27T07:48:46.253+0000 [INFO] [com.android.build.gradle.internal.profile.RecordingBuildListener] Task :nexusIQScan in app_bt Finished
2021-04-27T07:48:46.278+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Completing Build operation 'Task :nexusIQScan'
2021-04-27T07:48:46.278+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Build operation 'Task :nexusIQScan' completed
2021-04-27T07:48:46.278+0000 [INFO] [org.gradle.execution.plan.DefaultPlanExecutor] :nexusIQScan (Thread[Execution worker for ':' Thread 2,5,main]) completed. Took 0.123 secs.
2021-04-27T07:48:46.279+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :nexusIQScan failed
2021-04-27T07:48:46.279+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :app_bt:nexusIQScan completed, executed: false
2021-04-27T07:48:46.279+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :cardmodule:nexusIQScan completed, executed: false
2021-04-27T07:48:46.279+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :download:nexusIQScan completed, executed: false
2021-04-27T07:48:46.279+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :feedback:nexusIQScan completed, executed: false
2021-04-27T07:48:46.279+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :fileprocessing:nexusIQScan completed, executed: false
2021-04-27T07:48:46.280+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :loginmodule:nexusIQScan completed, executed: false
2021-04-27T07:48:46.280+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :moduleads:nexusIQScan completed, executed: false
2021-04-27T07:48:46.280+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :moduleanalytics:nexusIQScan completed, executed: false
2021-04-27T07:48:46.280+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :network:nexusIQScan completed, executed: false
2021-04-27T07:48:46.280+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :pdfdirector:nexusIQScan completed, executed: false
2021-04-27T07:48:46.280+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :purger:nexusIQScan completed, executed: false
2021-04-27T07:48:46.280+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :rewardsmodule:nexusIQScan completed, executed: false
2021-04-27T07:48:46.280+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :texttospeech:nexusIQScan completed, executed: false
2021-04-27T07:48:46.280+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :uikit:nexusIQScan completed, executed: false
2021-04-27T07:48:46.280+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :virocore:nexusIQScan completed, executed: false
2021-04-27T07:48:46.280+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :webviewads:nexusIQScan completed, executed: false
2021-04-27T07:48:46.280+0000 [DEBUG] [org.gradle.execution.plan.DefaultExecutionPlan] Node :nexusIQScan completed, executed: true
2021-04-27T07:48:46.280+0000 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Execution worker for ':' Thread 2: released lock on :
2021-04-27T07:48:46.281+0000 [DEBUG] [org.gradle.internal.work.DefaultWorkerLeaseService] Worker lease root.1.14 completed (1 worker(s) in use)
2021-04-27T07:48:46.281+0000 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Execution worker for ':' Thread 2: released lock on root.1.14
2021-04-27T07:48:46.281+0000 [DEBUG] [org.gradle.execution.plan.DefaultPlanExecutor] Execution worker [Thread[Execution worker for ':' Thread 2,5,main]] finished, busy: 0.123 secs, idle: 0.02 secs
2021-04-27T07:48:46.281+0000 [DEBUG] [org.gradle.execution.plan.DefaultPlanExecutor] Execution worker [Thread[Execution worker for ':',5,main]] finished, busy: 0.0 secs, idle: 0.138 secs
2021-04-27T07:48:46.281+0000 [DEBUG] [org.gradle.execution.plan.DefaultPlanExecutor] Execution worker [Thread[Execution worker for ':' Thread 4,5,main]] finished, busy: 0.0 secs, idle: 0.138 secs
2021-04-27T07:48:46.281+0000 [DEBUG] [org.gradle.execution.plan.DefaultPlanExecutor] Execution worker [Thread[Execution worker for ':' Thread 3,5,main]] finished, busy: 0.0 secs, idle: 0.138 secs
2021-04-27T07:48:46.281+0000 [DEBUG] [org.gradle.execution.plan.DefaultPlanExecutor] Execution worker [Thread[Daemon worker,5,main]] finished, busy: 0.0 secs, idle: 0.138 secs
2021-04-27T07:48:46.282+0000 [DEBUG] [org.gradle.execution.plan.DefaultPlanExecutor] Execution worker [Thread[Execution worker for ':' Thread 5,5,main]] finished, busy: 0.0 secs, idle: 0.138 secs
2021-04-27T07:48:46.293+0000 [DEBUG] [org.gradle.execution.taskgraph.DefaultTaskExecutionGraph] Timing: Executing the DAG took 0.185 secs
2021-04-27T07:48:46.297+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Completing Build operation 'Run tasks'
2021-04-27T07:48:46.297+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationExecutor] Build operation 'Run tasks' completed
.
.
2021-04-27T07:48:46.484+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] FAILURE: Build failed with an exception.
2021-04-27T07:48:46.484+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2021-04-27T07:48:46.484+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * What went wrong:
2021-04-27T07:48:46.484+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Some problems were found with the configuration of task ':nexusIQScan' (type 'NexusIqScanTask').
2021-04-27T07:48:46.484+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > No value has been specified for property 'applicationId'.
2021-04-27T07:48:46.484+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > No value has been specified for property 'password'.
2021-04-27T07:48:46.484+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > No value has been specified for property 'serverUrl'.
2021-04-27T07:48:46.484+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > No value has been specified for property 'username'.
2021-04-27T07:48:46.484+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2021-04-27T07:48:46.484+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Try:
2021-04-27T07:48:46.485+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]  Run with --scan to get full insights.
2021-04-27T07:48:46.485+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 
2021-04-27T07:48:46.485+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Exception is:
2021-04-27T07:48:46.487+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] org.gradle.internal.execution.WorkValidationException: Some problems were found with the configuration of task ':nexusIQScan' (type 'NexusIqScanTask').
2021-04-27T07:48:46.488+0000 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] 	at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:71)

Hi @puash,

I was able to reproduce the error after making some minor changes so the "nexusIQScan" could run, even using the version 2.0.9 of this plugin.

```
Caused by: java.util.ConcurrentModificationException
        at org.gradle.api.internal.DefaultDomainObjectCollection$IteratorImpl.next(DefaultDomainObjectCollection.java:479)
        at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.findResolvedDependencies(DependenciesFinder.java:80)
        at org.sonatype.gradle.plugins.scan.common.DependenciesFinder.lambda$findModules$9(DependenciesFinder.java:108)
        at org.gradle.api.internal.DefaultMutationGuard$2.execute(DefaultMutationGuard.java:42)
        at org.gradle.internal.Actions.with(Actions.java:251)
```

I'm attaching the modified project here for reference: 
[gradle-sample-error-reproduced.zip](https://github.com/sonatype-nexus-community/scan-gradle-plugin/files/6424282/gradle-sample-error-reproduced.zip)



Thanks for sharing this with us :)

--- a/src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java
+++ b/src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java
@@ -73,7 +73,7 @@ public class DependenciesFinder
 
   public Set<ResolvedDependency> findResolvedDependencies(Project rootProject, boolean allConfigurations) {
     return new LinkedHashSet<>(rootProject.getAllprojects()).stream()
-        .flatMap(project -> project.getConfigurations().stream())
+        .flatMap(project -> new LinkedHashSet<>(project.getConfigurations()).stream())
         .filter(configuration -> isAcceptableConfiguration(configuration, allConfigurations))
         .flatMap(configuration -> getDependencies(rootProject, configuration,
             resolvedConfiguration -> resolvedConfiguration.getFirstLevelModuleDependencies().stream()))

--- a/src/test/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinderTest.java
+++ b/src/test/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinderTest.java
@@ -24,6 +24,7 @@ import com.sonatype.insight.scan.module.model.Artifact;
 import com.sonatype.insight.scan.module.model.Dependency;
 import com.sonatype.insight.scan.module.model.Module;
 
+import com.google.common.collect.Sets;
 import org.gradle.api.Project;
 import org.gradle.api.artifacts.*;
 import org.gradle.api.artifacts.dsl.DependencyHandler;
@@ -40,6 +41,7 @@ import static org.gradle.api.plugins.JavaPlugin.COMPILE_CLASSPATH_CONFIGURATION_
 import static org.gradle.api.plugins.JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME;
 import static org.gradle.api.plugins.JavaPlugin.TEST_COMPILE_CLASSPATH_CONFIGURATION_NAME;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.when;
 
 public class DependenciesFinderTest
@@ -147,6 +149,21 @@ public class DependenciesFinderTest
   @Test
   public void testFindResolvedDependencies_includeTestDependencies() {
     Project project = buildProject(TEST_COMPILE_CLASSPATH_CONFIGURATION_NAME, false);
+    Set<ResolvedDependency> result = finder.findResolvedDependencies(project, true);
+    assertThat(result).hasSize(1);
+  }
+
+  @Test
+  public void testFindResolvedDependencies_copyConfigurationAfterResolveException() {
+    Project project = spy(buildProject(COMPILE_CLASSPATH_CONFIGURATION_NAME, false));
+    ConfigurationContainer configurationContainer = spy(project.getConfigurations());
+    Configuration configuration = spy(configurationContainer.iterator().next());
+
+    when(configuration.getResolvedConfiguration()).thenThrow(ResolveException.class);
+    when(configurationContainer.stream()).thenReturn(Stream.of(configuration));
+    when(project.getConfigurations()).thenReturn(configurationContainer);
+    when(project.getAllprojects()).thenReturn(Sets.newHashSet(project));
+
     Set<ResolvedDependency> result = finder.findResolvedDependencies(project, true);
     assertThat(result).hasSize(1);
   }

